icelake/src/io_v2/functional_writer/prometheus.rs

```diff
@@ -0,0 +1,78 @@
+//! Prometheus layer for FileAppender.
+
+use arrow_array::RecordBatch;
+use arrow_schema::SchemaRef;
+use async_trait::async_trait;
+use prometheus::{
+    core::{AtomicU64, GenericCounter},
+    Histogram,
+};
+
+use crate::{
+    io_v2::{IcebergWriter, IcebergWriterBuilder},
+    Result,
+};
+
+#[derive(Clone)]
+pub struct WriterMetrics {
+    write_qps: GenericCounter<AtomicU64>,
+    write_latency: Histogram,
+}
+
+impl WriterMetrics {
+    pub fn new(write_qps: GenericCounter<AtomicU64>, write_latency: Histogram) -> Self {
+        Self {
+            write_qps,
+            write_latency,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct PrometheusWriterBuilder<B: IcebergWriterBuilder> {
```

Comment:
I think this is too simple, we should expose much more metrics.

Reply:
What else metrics we need here?🤔 It's easy to extend it. This layer is used to store the common metrics. For the specific metrics such partition num, I prepare to add a Metrics trait so that we can put it the specific writer such PartitionWriter. I will add it after this PR.